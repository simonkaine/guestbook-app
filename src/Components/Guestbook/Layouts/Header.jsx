import React from 'react';

export default function Header() {
    return (
        <header>
            <div style={{height: '100px', border: '1px solid black'}}>
                <p>Welcome...</p>
            </div>
        </header>
    )
}

