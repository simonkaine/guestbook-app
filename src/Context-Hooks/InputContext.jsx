import { useContext, createContext, useState } from "react";

const inputContext = createContext();
const InputProvider = ({children}) => {

    const [input, setInput] = useState([]);

    return (
        <inputContext.Provider value={{input, setInput}}>
            {children}
        </inputContext.Provider>
    )
}

const useInputs = () => {
    const context = useContext(inputContext)
    if(context === undefined) {
        throw new Error("Yoooo, things ain't working with your inputs dawg..")
    } return context
}

export { InputProvider, useInputs }