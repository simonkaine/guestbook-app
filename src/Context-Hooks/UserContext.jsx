import { useContext, createContext, useState } from "react";

const UserContext = createContext();
const UserProvider = ({children}) => {

    const [user, setUser] = useState('');

    return (
        <UserContext.Provider value={{user, setUser}}>
            {children}
        </UserContext.Provider>
    )
}

const useUser = () => {
    const context = useContext(UserContext)
    if(context === undefined) {
        throw new Error("Yoooo, things ain't working with your 'user' dawg..")
    } return context;
}

export { UserProvider, useUser }