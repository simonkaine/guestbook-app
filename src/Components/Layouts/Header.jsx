import React from 'react';
import { useUser } from '../../Context-Hooks/UserContext';

export default function Header() {
    const { user } = useUser();

    return (
        <header>
            <div style={{height: '100px', border: '1px solid black', display: 'flex', justifyContent: 'flex-end'}}>
            {user ? (
                <h1 style={{margin: '0 20px 0 0'}}>Hello, {user}!</h1>
            ) : ( 
                <h1 style={{margin: '0 20px 0 0'}}>Welcome to Hotel california!</h1>
            )}
            </div>
        </header>
    )
}

