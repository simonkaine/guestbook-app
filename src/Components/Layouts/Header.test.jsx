import React from 'react'
import { render, screen } from '@testing-library/react'
import Header from './Header'
import { UserProvider } from '../../Context-Hooks/UserContext'

it('Should render the header', async () => {
    const {container} = render(

        <UserProvider>
            <Header />
        </UserProvider>

    )
    const header = await screen.findByText(/Welcome to Hotel california!/i)
    expect(header).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})