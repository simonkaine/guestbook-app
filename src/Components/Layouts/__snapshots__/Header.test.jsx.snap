// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the header 1`] = `
<div>
  <header>
    <div
      style="height: 100px; border: 1px solid black; display: flex; justify-content: flex-end;"
    >
      <h1
        style="margin: 0px 20px 0px 0px;"
      >
        Welcome to Hotel california!
      </h1>
    </div>
  </header>
</div>
`;
